controllers/routemonitor/routemonitor_supplement.go

```diff
@@ -4,34 +4,34 @@ import (
 	"context"
 
 	"github.com/openshift/route-monitor-operator/api/v1alpha1"
+	"github.com/openshift/route-monitor-operator/pkg/const/blackbox"
+	"github.com/openshift/route-monitor-operator/pkg/util/reconcile"
 	utilreconcile "github.com/openshift/route-monitor-operator/pkg/util/reconcile"
 )
 
-func (r *RouteMonitorReconciler) EnsureBlackBoxExporterResourcesExists(ctx context.Context) error {
-	if err := r.EnsureBlackBoxExporterDeploymentExists(ctx); err != nil {
-		return err
-	}
-	// Creating Service after because:
-	//
-	// A Service should not point to an empty target (Deployment)
-	if err := r.EnsureBlackBoxExporterServiceExists(ctx); err != nil {
-		return err
-	}
-	return nil
+//go:generate mockgen -source $GOFILE -destination ../../pkg/util/test/generated/mocks/$GOPACKAGE/blackboxexporter.go -package $GOPACKAGE BlackboxExporter
+type BlackboxExporter interface {
+	EnsureBlackBoxExporterResourcesExist() error
+	EnsureBlackBoxExporterResourcesAbsent() (reconcile.Result, error)
+	ShouldDeleteBlackBoxExporterResources() (blackbox.ShouldDeleteBlackBoxExporter, error)
+}
+
+func (r *RouteMonitorReconciler) EnsureBlackBoxExporterResourcesExists(ctx context.Context, blackboxExporter BlackboxExporter) error {
+	return blackboxExporter.EnsureBlackBoxExporterResourcesExist()
```

Comment:
if this just wraps the call, move it to use the actual struct and mock it with an interface

Reply:
ack